examples/fairmotion_interface.py

```diff
@@ -568,31 +587,409 @@ def belongs_to(self, obj_id: int) -> bool:
 
         return False
 
+    def push_action_order(self) -> None:
+        """
+        Place an action order at the end of the order queue to be consumed by process_action_order().
+        """
+        queue = self.order_queue
+
+        def snap_to_NM(point) -> mn.Vector3:
+            pf = self.sim.pathfinder
+
+            # drop location below NavMesh to guarantee bottom floor snap
+            point = mn.Vector3(point) + mn.Vector3(0.0, -2.0, 0.0)
+
+            # snap point to NavMesh
+            point = pf.snap_point(point)
+
+            return point
+
+        action_orders_library = {
```

Comment:
Can this be `self.action_orders_library` and setup in advance (e.g. in `__init__()`)?
Then external modifications will be taken into account here.